develop/networking.md

```diff
@@ -0,0 +1,255 @@
+client using packets, you only need to communicate with the game client like you're always running on a dedicated
+server.
+
+Connection to a server over LAN or single-player can also be treated like the server is a remote, dedicated server; so
+your game client can't directly access the server instance.
+
+## An Introduction to Networking {#an-introduction-to-networking}
+
+### Defining a Payload {#defining-a-payload}
+
+First, you need to define an `Identifier` used to identify our packet's payload. For this example our identifier will be
+`fabric-docs-reference:summon_lightning`.
+
+We can define this identifier in our **common initializer** for easy access.
+
+@[code lang=java transclude={16-16}](@/reference/latest/src/main/java/com/example/docs/networking/basic/FabricDocsReferenceNetworkingBasic.java)
+
+Now, let's define the payload that will be sent with the packet. This payload will contain the position (a `BlockPos`)
+of the item user.
+
+This can be done by creating a Java `Record` with a `BlockPos` parameter that implements `CustomPayload`.
+
+@[code lang=java transcludeWith=:::summon_Lightning_payload](@/reference/latest/src/main/java/com/example/docs/networking/basic/SummonLightningPayload.java)
+
+At the same time, we've defined:
+
+- A public static instance of `CustomPayload.Id` to uniquely identify this custom payload. We will be referencing this
+  ID in both our common and client code.
+
+  @[code lang=java transclude={10-10}](@/reference/latest/src/main/java/com/example/docs/networking/basic/SummonLightningPayload.java)
+
+- A public static instance of a `PacketCodec` so that the game knows how to serialize/deserialize the contents of the
+  packet.
+
+  @[code lang=java transclude={11-11}](@/reference/latest/src/main/java/com/example/docs/networking/basic/SummonLightningPayload.java)
+
+We have also overridden `getId` to return our payload ID.
+
+### Registering a Payload {#registering-a-payload}
+
+Before we send a packet with our custom payload, we need to register it.
+
+This can be done in our **common initializer** by `PayloadTypeRegistry.playS2C().register` which takes in a
+`CustomPayload.Id` and a `PacketCodec`.
+
+@[code lang=java transclude={22-22}](@/reference/latest/src/main/java/com/example/docs/networking/basic/FabricDocsReferenceNetworkingBasic.java)
+
+A similar method exists to register client-to-server payloads: `PayloadTypeRegistry.playC2S().register`.
+
+`S2C` and `C2S` are two common suffixes that mean _Server-to-Client_ and _Client-to-Server_ respectively.
+
+### Sending a Packet to the Client {#sending-a-packet-to-the-client}
+
+To send a packet with our custom payload, we can use `ServerPlayNetworking.send` which takes in a `ServerPlayerEntity`
+and a `CustomPayload`.
+
+Let's start by creating our Lightning Tater item. You can override `use` to add trigger an action when the item is used.
+In this case, let's send packets to all the players in the server.
```

Comment:
This statement is wrong. In the example code, the payload is only sent to players tracking the `BlockPos`. Tracking is a very important concept for efficient networking, it shouldn't be omitted.

Reply:
Noted. I believe using `world` would be better to match the description. Then I could add a statement about tracking and why it is important.